File: mobile-app/lib/models/livraison.dart
class Livraison {
  final int id;
  final String numeroLivraison;
  final String? origine;
  final String? destination;
  final String? dateDepart;
  final String? dateLivraison;
  final String statut;
  final Map<String, dynamic>? commande;

  const Livraison({
    required this.id,
    required this.numeroLivraison,
    this.origine,
    this.destination,
    this.dateDepart,
    this.dateLivraison,
    required this.statut,
    this.commande,
  });

  factory Livraison.fromJson(Map<String, dynamic> j) => Livraison(
        id: j['id'],
        numeroLivraison: j['numero_livraison'] ?? '',
        origine: j['origine'],
        destination: j['destination'],
        dateDepart: j['date_depart'],
        dateLivraison: j['date_livraison'],
        statut: j['statut'],
        commande: j['commande'],
      );
}

File: backend/app/Models/Livraison.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Str;

class Livraison extends Model
{
    public $timestamps = false;

    protected $fillable = [
        'numero_livraison',
        'commande_id',
        'transporteur_id',
        'origine',
        'destination',
        'destination_latitude',
        'destination_longitude',
        'date_depart',
        'date_livraison',
        'statut',
    ];

    protected static function booted(): void
    {
        static::creating(function (Livraison $livraison) {
            if (empty($livraison->numero_livraison)) {
                $livraison->numero_livraison = 'LIV-' . now()->format('Ymd') . '-' . strtoupper(Str::random(4));
            }
        });
    }

    public function commande()
    {
        return $this->belongsTo(Commande::class, 'commande_id');
    }

    public function transporteur()
    {
        return $this->belongsTo(Transporteur::class, 'transporteur_id');
    }
}
